apps/syntrest/src/main/java/no/nav/registre/syntrest/consumer/GitHubConsumer.java

```diff
@@ -0,0 +1,97 @@
+package no.nav.registre.syntrest.consumer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.ProxyProvider;
+
+import java.util.Objects;
+
+@Slf4j
+@Component
+public class GitHubConsumer {
+    /**
+     * The GitHub Consumer provides an interface to GitHub, primarily to retrieve package tags.
+     */
+
+    private final WebClient.Builder webClientBuilder;
+    private final String username;
+    private final String password;
+
+    public GitHubConsumer(
+            @Value("${github_username}") String username,
+            @Value("${github_password}") String password,
+            @Value("${github_url}") String githubUrl,
+            @Value("${proxy-url") String proxyUrl,
+            @Value("${proxy-port}") int proxyPort,
+            WebClient.Builder webClientBuilder
+    ) {
+
+        this.webClientBuilder = webClientBuilder.baseUrl(githubUrl);
+
+        if (!"local".equals(proxyUrl) && proxyPort != 0) {
+            HttpClient httpClient = HttpClient.create()
+                    .tcpConfiguration(tcpClient -> tcpClient.proxy(proxy -> proxy
+                            .type(ProxyProvider.Proxy.HTTP)
+                            .host(proxyUrl)
+                            .port(proxyPort)));
+
+            ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
+            this.webClientBuilder.clientConnector(connector);
+        }
+
+        this.username = username;
+        this.password = password;
+    }
```

Comment:
Hvorfor ikke bruke `private final WebClient webClient` i stedet? Kan sette at `this.webClient = webClientBuilder.build()` etter at baseurl og clientConnentor er satt.

Reply:
Samme kommentar igjen ;)

Reply:
Her blir GitHubConsumer opprettet som en bean, så webclient blir bare opprettet én gang. WebClient kan ikke være en bean i seg selv, siden det vil være forskjellige baseUrls